web/experimental/inject_embed.dart

```diff
@@ -0,0 +1,72 @@
+import 'dart:html';
+
+import 'package:html_unescape/html_unescape.dart';
+
+// Change src to 'embed-new.html?fw=true for local development
+const iFrameLocation =
+    'https://dartpad.dartlang.org/experimental/embed-new.html?fw=true';
+
+/// Replaces all code snippets marked with the 'dartpad-embed' class with an
+/// instance of DartPad.
+void main() {
+  var hosts = querySelectorAll('.dartpad-embed');
+  for (var host in hosts) {
+    _injectEmbed(host);
+  }
+}
+
+/// Replaces [host] with an instance of DartPad as an embedded iframe.
+///
+/// Code snippets are assumed to be a div containing `pre` and `code` tags:
+///
+/// <div class="dartpad-embed">
+///   <pre>
+///     <code>
+///       void main() => print("Hello, World!");
+///     </code>
+///   </pre>
+/// </div>
+void _injectEmbed(DivElement host) {
+  if (host.children.length != 1) {
+    return;
+  }
+
+  var preElement = host.children.first;
+  if (preElement.children.length != 1) {
+    return;
+  }
+
+  var codeElement = preElement.children.first;
+  var code = HtmlUnescape().convert(codeElement.innerHtml);
+  if (code.isEmpty) {
+    return;
+  }
+  InjectedEmbed(host, code);
+}
+
+/// Clears children in [host], instantiates an iframe, and sends it a message
+/// with the source code when it's ready
+class InjectedEmbed {
+  final DivElement host;
+  final String code;
+
+  InjectedEmbed(this.host, this.code) {
+    _init();
+  }
+
+  Future _init() async {
+    host.children.clear();
+
+    var iframe = IFrameElement()..setAttribute('src', iFrameLocation);
+    iframe.style.setProperty('border', '1px solid #ccc');
```

Comment:
Better to let the containing page set this?

Reply:
Maybe... the intention was to avoid the iframe looking strange out of the box. If we take it out we would probably want to add documentation to say that we replace the code with an iframe and you need to update your css to add this line 

<img width="793" alt="Screen Shot 2019-05-22 at 10 45 31 AM" src="https://user-images.githubusercontent.com/1145719/58196248-c0919280-7c7e-11e9-8218-6ff7b27c0204.png">